File: test_example.js
#!/usr/bin/env node
import { formatHtml } from './src/html-formatter.js';

const code = `function hello(name) {
  // Greet the user
  console.log("Hello, " + name + "!");
  return true;
}

hello("World");`;

async function test() {
  console.log('Testing HTML formatter...\n');

  // Test 1: Simple inline HTML (no standalone)
  console.log('Test 1: Inline HTML');
  const html1 = await formatHtml(code, 'javascript', {
    standalone: false,
    theme: 'github-dark',
  });
  console.log(html1.substring(0, 200) + '...\n');

  // Test 2: Standalone HTML
  console.log('Test 2: Standalone HTML');
  const html2 = await formatHtml(code, 'javascript', {
    standalone: true,
    theme: 'monokai',
    lineNumbers: true,
  });
  console.log(html2.substring(0, 400) + '...\n');

  console.log('Tests completed!');
}

test().catch(console.error);

File: test_example.py
#!/usr/bin/env python3
"""A simple Python example for testing"""

def greet(name):
    # Greet the user
    print(f"Hello, {name}!")
    return True

class Greeter:
    """A greeter class"""
    def __init__(self, name):
        self.name = name

    def say_hello(self):
        print(f"Hi, {self.name}!")

if __name__ == "__main__":
    greet("World")
    greeter = Greeter("Python")
    greeter.say_hello()
